pkg/source/kinesis/kinesis_source_test.go

```diff
@@ -339,15 +344,104 @@ func TestKinesisSourceHCL(t *testing.T) {
+			resultConf, ok := result.(*Configuration)
+			if !ok {
+				t.Fatal("result is not of type pointer to Configuration")
 			}
+
+			assert.Equal(resultConf.StreamName, tt.Expected.StreamName)
+			assert.Equal(resultConf.Region, tt.Expected.Region)
+			assert.Equal(resultConf.AppName, tt.Expected.AppName)
+			assert.Equal(resultConf.RoleARN, tt.Expected.RoleARN)
+			assert.Equal(resultConf.StartTimestamp, tt.Expected.StartTimestamp)
+			assert.Equal(resultConf.ConcurrentWrites, tt.Expected.ConcurrentWrites)
+			assert.Equal(resultConf.ReadThrottleDelayMs, tt.Expected.ReadThrottleDelayMs)
+			assert.Equal(resultConf.ShardCheckFreqSeconds, tt.Expected.ShardCheckFreqSeconds)
+			assert.Equal(resultConf.LeaderActionFreqSeconds, tt.Expected.LeaderActionFreqSeconds)
+
+			if !tt.ClientNameUUID {
+				assert.Equal(resultConf.ClientName, tt.Expected.ClientName)
+			} else {
+				_, err := uuid.Parse(resultConf.ClientName)
+				assert.Nil(err)
+			}
+
 		})
 	}
 }
 
+func TestClientNamePassThrough(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	assert := assert.New(t)
+
+	filename := filepath.Join(assets.AssetsRootDir, "test", "config", "configs", "empty.hcl")
+	t.Setenv("SNOWBRIDGE_CONFIG_FILE", filename)
+
+	clientName := "testt_client_name"
+	kinesisClient := testutil.GetAWSLocalstackKinesisClient()
+	dynamodbClient := testutil.GetAWSLocalstackDynamoDBClient()
+	streamName := "kinesis-source-config-integration-4"
+	createErr := testutil.CreateAWSLocalstackKinesisStream(kinesisClient, streamName, 1)
+	if createErr != nil {
+		t.Fatal(createErr)
+	}
+	defer testutil.DeleteAWSLocalstackKinesisStream(kinesisClient, streamName)
+
+	appName := "kinesisSourceIntegration"
+	ddbErr := testutil.CreateAWSLocalstackDynamoDBTables(dynamodbClient, appName)
+	if ddbErr != nil {
+		t.Fatal(ddbErr)
+	}
+	defer testutil.DeleteAWSLocalstackDynamoDBTables(dynamodbClient, appName)
+
+	// Construct the config
+	c, err := config.NewConfig()
+	assert.NotNil(c)
+	if err != nil {
+		t.Fatalf("function NewConfig failed with error: %q", err.Error())
+	}
+
+	configBytesToMerge := []byte(fmt.Sprintf(`
+    stream_name = "%s"
+    region      = "%s"
+    app_name    = "%s"
+    client_name = "%s"
+`, streamName, testutil.AWSLocalstackRegion, appName, clientName))
+
+	parser := hclparse.NewParser()
+	fileHCL, diags := parser.ParseHCL(configBytesToMerge, "placeholder")
+	if diags.HasErrors() {
+		t.Fatalf("failed to parse config bytes")
+	}
+
+	c.Data.Source.Use.Name = "kinesis"
+	c.Data.Source.Use.Body = fileHCL.Body
+	decoderOpts := &config.DecoderOptions{
+		Input: c.Data.Source.Use.Body,
+	}
+	plug := adapterGenerator(configFunctionGeneratorWithInterfaces(kinesisClient, dynamodbClient, "00000000000"))
+	result, err := c.CreateComponent(plug, decoderOpts)
+	assert.NotNil(result)
+	if err != nil {
+		t.Fatalf("failed to CreateComponent with error: %q", err.Error())
+	}
+
+	// workaround since clientName is unexported by Kinsumer
```

Comment:
Not asking you to implement anything, but have a think about other possible options to verify the feature, and when we catch up let's think about what the simplest options might be.

Reply:
Followup - we discussed different ways of thinking on the call.